File: Classes/Helpers/CenteredTextLayer.m
//
//  CenteredTextLayer.m
//  Simple Comic
//
//  Created by Franco Solerio on 17/04/2020.
//  Copyright © 2020 Dancing Tortoise Software. All rights reserved.
//

#import "CenteredTextLayer.h"

@implementation CenteredTextLayer

- (void)drawInContext:(CGContextRef)ctx {
	CGFloat height = self.bounds.size.height;
	CGFloat fontSize = self.fontSize;
	CGFloat yDiff = (height-fontSize)/2 - fontSize/10;
	
	CGContextSaveGState(ctx);
	CGContextTranslateCTM(ctx, 0, -yDiff);
	[super drawInContext:ctx];
	CGContextRestoreGState(ctx);
}

@end

File: Classes/Helpers/CenteredTextLayer.swift
//
//  CenteredTextLayer.swift
//  Simple Comic
//
//  Created by J-rg on 09.12.20.
//  Copyright © 2020 Dancing Tortoise Software. All rights reserved.
//

import QuartzCore

@objc
class CenteredTextLayer: CATextLayer {
	
	override func draw(in ctx: CGContext) {
		let height = bounds.size.height
		let fontSize = self.fontSize
		let yDiff = (height - fontSize) / 2 - fontSize / 10
		
		ctx.saveGState()
		ctx.translateBy(x: 0, y: -yDiff)
		super.draw(in: ctx)
		ctx.restoreGState()
	}
}
